tests/test_external_search.py

```diff
@@ -414,226 +352,156 @@ def filters_to(start, finish):
-        # make comparisons easier.
-        actual_ids = []
-        actual_titles = []
-        for work in actual:
-            actual_titles.append(work.title)
-            actual_ids.append(work.id)
-
-        expect_ids = []
-        expect_titles = []
-        for work in expect:
-            expect_titles.append(work.title)
-            expect_ids.append(work.id)
-
-        eq_(
-            expect, actual,
-            "%r did not find %d works\n (%s/%s).\nInstead found %d\n (%s/%s)" % (
-                description,
-                len(expect), ", ".join(map(str, expect_ids)),
-                    ", ".join(expect_titles),
-                len(actual), ", ".join(map(str, actual_ids)),
-                    ", ".join(actual_titles)
-            )
-        )
-
-    def _expect_results(self, expect, *query_args, **kwargs):
-        """Helper function to call query() and verify that it
-        returns certain work IDs.
-
-        :param should_be_ordered: If this is True (the default), then the
-        assertion will only succeed if the search results come in in
-        the exact order specified in `works`. If this is False, then
-        those exact results must come up, but their order is not
-        what's being tested.
-        """
-        if isinstance(expect, Work):
-            expect = [expect]
-
-        should_be_ordered = kwargs.pop('ordered', True)
-
-        results = self.search.query_works(*query_args, debug=True, **kwargs)
-        actual = self._db.query(Work).filter(Work.id.in_(results)).all()
-        if should_be_ordered:
-            # Put the Work objects in the same order as the IDs returned
-            # in `results`.
-            works_by_id = dict()
-            for w in actual:
-                works_by_id[w.id] = w
-            actual = [
-                works_by_id[result] for result in results
-                if result in works_by_id
-            ]
-
-        self._assert_works(query_args, expect, actual, should_be_ordered)
-
-
-class TestExternalSearchWithWorks(EndToEndExternalSearchTest):
+class TestExternalSearchWithWorks(EndToEndSearchTest):
     """These tests run against a real search index with works in it.
     The setup is very slow, so all the tests are in the same method.
     Don't add new methods to this class - add more tests into test_query_works,
     or add a new test class.
     """
 
-    def setup(self):
-        super(TestExternalSearchWithWorks, self).setup()
+    def populate_works(self):
         _work = self.default_work
 
-        if self.search:
+        self.moby_dick = _work(
+            title="Moby Dick", authors="Herman Melville", fiction=True,
+        )
+        self.moby_dick.presentation_edition.subtitle = "Or, the Whale"
+        self.moby_dick.presentation_edition.series = "Classics"
+        self.moby_dick.summary_text = "Ishmael"
+        self.moby_dick.presentation_edition.publisher = "Project Gutenberg"
```

Comment:
Most of the changes in this file are removing an indent level because it's no longer necessary to check `self.search` before creating works. `populate_works` is only called if a search client could be created.